assets/js/modules/ads/components/setup/SetupMainPAX.js

```diff
@@ -0,0 +1,230 @@
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import AdsIcon from '../../../../../svg/graphics/ads.svg';
+import SetupForm from './SetupForm';
+import SupportLink from '../../../../components/SupportLink';
+import AdBlockerWarning from '../common/AdBlockerWarning';
+import { SpinnerButton } from 'googlesitekit-components';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { CORE_LOCATION } from '../../../../googlesitekit/datastore/location/constants';
+import { ADWORDS_SCOPE, MODULES_ADS } from '../../datastore/constants';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import useQueryArg from '../../../../hooks/useQueryArg';
+import { addQueryArgs } from '@wordpress/url';
+import PAXEmbeddedApp from '../PAXEmbeddedApp';
+const { useSelect, useDispatch, useRegistry } = Data;
+
+export default function SetupMainPAX( { finishSetup } ) {
+	const [ showPaxApp, setShowPaxApp ] = useQueryArg( 'pax' );
+
+	const registry = useRegistry();
+
+	const isAdBlockerActive = useSelect( ( select ) =>
+		select( CORE_USER ).isAdBlockerActive()
+	);
+	const hasAdwordsScope = useSelect( ( select ) =>
+		select( CORE_USER ).hasScope( ADWORDS_SCOPE )
+	);
+
+	const redirectURL = addQueryArgs( global.location.href, {
+		pax: 1,
+		// The `notification: 'ads'` query value is arbitrary and serves
+		// two purposes:
+		//
+		// 1. To ensure that `authentication_success` isn't appended
+		//    when returning from OAuth.
+		// 2. To guarantee it doesn't match any existing notifications
+		//    in the `BannerNotifications` component, thus preventing
+		//    any unintended displays.
+		notification: 'ads',
+	} );
+
+	const oAuthURL = useSelect( ( select ) =>
+		select( CORE_USER ).getConnectURL( {
+			additionalScopes: [ ADWORDS_SCOPE ],
+			redirectURL,
+		} )
+	);
+	const isNavigatingToOAuthURL = useSelect( ( select ) => {
+		if ( ! oAuthURL ) {
+			return false;
+		}
+
+		return select( CORE_LOCATION ).isNavigatingTo( oAuthURL );
+	} );
+
+	const { navigateTo } = useDispatch( CORE_LOCATION );
+	const { setPaxConversionID, setExtCustomerID, submitChanges } =
+		useDispatch( MODULES_ADS );
+
+	const onLaunch = useCallback(
+		async ( app ) => {
+			/* eslint-disable sitekit/acronym-case */
+			// Disabling rule because function and property names
+			// are expected in current format by PAX API.
+			const customerData = await app
+				.getServices()
+				.accountService.getAccountId( {} );
+			const conversionTrackingData = await app
+				.getServices()
+				?.conversionTrackingIdService.getConversionTrackingId( {} );
+
+			if (
+				customerData.externalCustomerId &&
+				conversionTrackingData.conversionTrackingId
+			) {
+				setExtCustomerID( customerData.externalCustomerId );
+				setPaxConversionID(
+					conversionTrackingData.conversionTrackingId
+				);
+				/* eslint-enable sitekit/acronym-case */
+
+				const { error } = await submitChanges();
```

Comment:
This condition results in an automatic submit since the customer ID and CTID can exist even when there are no campaigns (which is odd, but possible).

At this stage, we don't have the campaign created signal to really know if a campaign has been created or not so this really needs to be triggered by the user. Even when we do have that signal, it should enhance the experience, but still shouldn't be an automatic submit.